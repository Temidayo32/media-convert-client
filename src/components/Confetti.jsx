import React, { useEffect } from 'react';
import Particles, { initParticlesEngine } from '@tsparticles/react';
// import { loadSlim } from "@tsparticles/slim";
import { loadFull } from "tsparticles";

function Confetti() {
    useEffect(() => {
        const initializeParticles = async () => {
          await initParticlesEngine(async (engine) => {
            await loadFull(engine);
          });
        };
    
        initializeParticles();
      }, []);

  const options = {
    background: {
      color: {
        value: "transparent",
      },
    },
    fullScreen: {
        enable: true,
        zIndex: 1,
    },
    emitters: {
        position: {
          x: 50,
          y: 100
        },
        rate: {
          quantity: 5,
          delay: 0.15
        }
      },
    particles: {
      number: {
        value: 100,
      },
      color: {
        value: ["#4285F4", "#DB4437", "#0F9D58", "#5C6BC0"],
      },
      shape: {
        type: ['circle', 'square', 'triangle'],
      },
      size: {
        value: 6,
        random: true,
      },
      move: {
        decay: 0.05,
        direction: "top",
        enable: true,
        gravity: {
          enable: true
        },
        outModes: {
          top: "none",
          default: "destroy"
        },
        speed: {
          min: 50,
          max: 100
        }
      },
      opacity: {
        value: 1
      },
      rotate: {
        value: {
          min: 0,
          max: 360
        },
        direction: "random",
        animation: {
          enable: true,
          speed: 30
        }
      },
      tilt: {
        direction: "random",
        enable: true,
        value: {
          min: 0,
          max: 360
        },
        animation: {
          enable: true,
          speed: 30
        }
      },
      size: {
        value: 3,
        animation: {
          enable: true,
          startValue: "min",
          count: 1,
          speed: 16,
          sync: true
        }
      },
      roll: {
        darken: {
          enable: true,
          value: 25
        },
        enlighten: {
          enable: true,
          value: 25
        },
        enable: true,
        speed: {
          min: 5,
          max: 15
        }
      },
      wobble: {
        distance: 30,
        enable: true,
        speed: {
          min: -7,
          max: 7
        }
      },
      responsive: [
        {
          maxWidth: 1024,
          options: {
            particles: {
              move: {
                speed: {
                  min: 33,
                  max: 66
                }
              }
            }
          }
        }
      ]
    },
    preset: "confetti",
  };

  const particlesLoaded = (container) => {
    console.log(container);
  };

  return (
    <div className="absolute top-0 left-0 w-full h-full z-10">
        <Particles
            id="tsparticles"
            options={options}
            particlesLoaded={particlesLoaded}
    />
    </div>
  );
}

export default Confetti;
