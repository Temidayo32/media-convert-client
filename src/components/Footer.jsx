import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';

const Footer = () => {
  const [videoConversions, setVideoConversions] = useState([]);
  const [imageConversions, setImageConversions] = useState([]);

  useEffect(() => {
    fetch('/conversions.json')
      .then(response => response.json())
      .then(data => {
        setVideoConversions(data.videos);
        setImageConversions(data.images);
      })
      .catch(error => console.error('Error fetching conversions:', error));
  }, []);

  return (
    <footer className="bg-gray-900 text-white py-12">
      <div className="container mx-auto px-4">
        <div className="flex flex-col sm:flex-row gap-6 justify-between text-center">
          {/* Company Logo and Name Section */}
          <div className="w-full sm:w-2/6 lg:w-1/3 flex items-center sm:items-start justify-center sm:justify-normal mb-4 lg:mb-0">
            <img src="/medialogo.svg" alt="Company Logo" className="w-8 h-8 lg:w-12 lg:h-12 mr-2 mb-4" />
            <span className="text-base lg:text-lg font-bold text-teal-400 mb-4 sm:mt-1 lg:mt-2">Media Convert</span>
          </div>

          {/* Product Features and Conversion Tools Section */}
          <div className="w-full sm:w-4/6 lg:w-2/3 flex flex-col sm:flex-row justify-between">
            {/* Product Features Column */}
            <div className="w-full sm:w-1/2">
              <h4 className="text-base lg:text-lg font-bold mb-4 text-teal-400">Product Features</h4>
              <ul>
                <li className="mb-2"><a href="/video-converter" className="text-gray-400 text-xs md:text-sm lg:text-base hover:text-white">Video converter</a></li>
                <li className="mb-2"><a href="#" className="text-gray-400 text-xs md:text-sm lg:text-base hover:text-white">Image converter </a></li>  
                {/* <sup className="border border-orange-500 rounded-full py-1 text-xs px-1">coming soon</sup> */}
                <li className="mb-2"><a href="#" className="text-gray-400 text-xs md:text-sm lg:text-base hover:text-white">Document converter <sup className="border border-orange-500 rounded-full py-1 text-xs px-1">coming soon</sup></a></li>
              </ul>
            </div>

            {/* Video Conversion Tools Column */}
            <div className="w-full sm:w-1/2">
              <h4 className="text-base lg:text-lg font-bold mb-4 text-teal-400">Video Conversion Tools</h4>
              <ul>
                {videoConversions.map((conversion, index) => (
                  <li key={index} className="mb-2">
                    <Link to={`/video-converter/${conversion.format.toLowerCase()}`} className="text-xs md:text-sm lg:text-base text-gray-400 hover:text-white">{conversion.format} converter</Link>
                  </li>
                ))}
              </ul>
            </div>
          </div>

          {/* Image Conversion Tools Column */}
          <div className="w-full sm:w-1/2 mt-6 sm:mt-0">
            <h4 className="text-base lg:text-lg font-bold mb-4 text-teal-400">Image Conversion Tools</h4>
            <ul>
              {imageConversions.map((conversion, index) => (
                <li key={index} className="mb-2">
                  <Link to={`/image-converter/${conversion.format.toLowerCase()}`} className="text-xs md:text-sm lg:text-base text-gray-400 hover:text-white">{conversion.format} converter</Link>
                </li>
              ))}
            </ul>
          </div>
        </div>

        {/* Copyright Section */}
        <p className="text-sm text-center text-gray-400 mt-6">&copy; 2024 Media Convert. All Rights Reserved.</p>
      </div>
    </footer>
  );
};

export default Footer;
