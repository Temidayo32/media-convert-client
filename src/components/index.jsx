export {default as AccountSuccess} from './home/AccountSuccess';
export {default as Card } from './home/Card';
export {default as Header} from './home/Header';
export { default as HeaderOptions} from './home/HeaderOptions';
export {default as HeroSection} from './home/HeroSection';
export {default as Login} from './home/Login';
export {default as LoginWithEmail } from './home/LoginWithEmail';
export {default as Signup} from './home/Signup';
export {default as SignUpWithEmail} from './home/SignUpWithEmail';
export {default as SignUpWithGoogle } from './home/SignUpWithGoogle';
export {default as PasswordReset} from './home/PasswordReset';
export {default as EmailVerification} from "./home/EmailVerification";
export {default as EmailVerificationRedirect} from './home/EmailVerificationRedirect';
export { default as Menu } from './home/Menu';
export {default as EmailVerificationMenu} from './home/EmailVerificationMenu';
export { default as EmailAdminVerication } from './home/EmailAdminVerication';
export {default as VideoSection} from './home/VideoSection';
export {default as Footer} from './Footer';
export {default as Signupmodal} from './home/Signupmodal'
export {default as UploadVideo} from './video/UploadVideo';
export {default as UploadOptions} from './video/UploadOptions';
export {default as OneDrivePicker} from './video/OneDrivePicker';
export {default as Conversion} from './video/Conversion';
export {default as AdvancedOptions} from './video/AdvancedOptions';
export {default as Dashboard} from './dashboard/Dashboard';
export {default as Profile } from './dashboard/Profile';
export {default as RecentTasks} from './dashboard/RecentTasks';
export {default as ChangeEmail } from './dashboard/ChangeEmail';
export {default as ChangePassword } from './dashboard/ChangePassword';
export {default as DeleteAccount } from './dashboard/DeleteAccount';
export {default as DownloadOptions } from './video/DownloadOptions';
export {default as VideoHero } from './videoPage/VideoHero';
export {default as HowToConvert } from './videoPage/HowToConvert';
export {default as FaqItem } from './videoPage/FaqItem';
export {default as FaqSection } from './videoPage/FaqSection';
export { default as Audience } from './home/Audience';
export { default as AudienceCard } from './home/AudienceCard';
export { default as UploadImage } from './image/UploadImage';
export { default as EditImage } from './image/EditImage';
export { default as CropImage } from './image/CropImage';
export {default as ImageThumbnails} from './image/ImageThumbnails';