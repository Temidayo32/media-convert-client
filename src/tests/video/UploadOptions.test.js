import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { UploadOptions } from '../../components';
import { BrowserRouter as Router } from 'react-router-dom';


/**
 * Test Suite for UploadOptions Component
 
 * 1. Render the component without crashing.
 * 2. Test the file upload functionality.
 * 3. Test the Dropbox selection functionality.
 * 4. Test the Google Drive selection functionality.
 */

jest.mock('react-dropbox-chooser', () => ({ success, children }) => (
  <div data-testid="mock-dropbox-chooser" onClick={() => success([{ link: 'dropbox-file-link' }])}>
    {children}
  </div>
));

global.setImmediate = (callback) => setTimeout(callback, 0);

console.error = jest.fn();
console.log = jest.fn();

describe('UploadOptions Component', () => {
  afterEach(() => {
    jest.clearAllMocks(); // Clear mock function calls after each test
  });

  test('renders correctly', () => {
    const handleFileUpload = jest.fn();
    const handleOpenPicker = jest.fn();
    const onSuccess = jest.fn();
    const onCancel = jest.fn();
    const handleFileSelected = jest.fn();

    const { getByText, getByTestId, asFragment } = render(
      <Router>
        <UploadOptions
          handleFileUpload={handleFileUpload}
          handleOpenPicker={handleOpenPicker}
          onSuccess={onSuccess}
          onCancel={onCancel}
          handleFileSelected={handleFileSelected}
        />
      </Router>
    );

    expect(getByText('From Device')).toBeInTheDocument();
    expect(getByTestId('mock-dropbox-chooser')).toBeInTheDocument();
    expect(getByText('From Google Drive')).toBeInTheDocument();

    // Snapshot test
    expect(asFragment()).toMatchSnapshot();
  });

  test('handles file upload', () => {
    const handleFileUpload = jest.fn();

    const { container, asFragment } = render(
      <Router>
        <UploadOptions handleFileUpload={handleFileUpload} />
      </Router>
    );

    const uploadInput = container.querySelector('#upload-file');
    fireEvent.change(uploadInput, {
      target: {
        files: [new File(['(⌐□_□)'], 'large_video.mp4', { type: 'video/mp4' })],
      },
    });

    expect(handleFileUpload).toHaveBeenCalled();

    // Snapshot test
    expect(asFragment()).toMatchSnapshot();
  });

  test('handles Dropbox selection', () => {
    const onSuccess = jest.fn();
    const onCancel = jest.fn();

    const { getByTestId, asFragment } = render(
      <Router>
        <UploadOptions onSuccess={onSuccess} onCancel={onCancel} />
      </Router>
    );

    fireEvent.click(getByTestId('mock-dropbox-chooser'));

    expect(onSuccess).toHaveBeenCalledWith([{ link: 'dropbox-file-link' }]);

    // Snapshot test
    expect(asFragment()).toMatchSnapshot();
  });

  test('handles Google Drive selection', () => {
    // Mock functions
    const handleOpenPicker = jest.fn();
    const onSuccess = jest.fn();
    const onCancel = jest.fn();
    const { getByText, asFragment } = render(
      <Router>
        <UploadOptions handleOpenPicker={handleOpenPicker} onSuccess={onSuccess} onCancel={onCancel} />
      </Router>
    );

    // Find the Google Drive link
    const googleDriveLink = getByText('From Google Drive');

    expect(googleDriveLink).toBeInTheDocument();
    fireEvent.click(googleDriveLink);
    expect(handleOpenPicker).toHaveBeenCalled();
    expect(handleOpenPicker).toHaveBeenCalledTimes(1);

    // Snapshot test
    expect(asFragment()).toMatchSnapshot();
  });
});
