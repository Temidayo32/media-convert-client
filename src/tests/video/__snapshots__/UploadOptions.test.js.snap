// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadOptions Component handles Dropbox selection 1`] = `
<DocumentFragment>
  <div
    class="absolute text-2xl top-full left-0 w-full bg-teal-500 text-center text-white rounded-lg my-px shadow-lg"
  >
    <ul>
      <label
        class="cursor-pointer"
        for="upload-file"
      >
        <li
          class="p-4 pl-8 hover:bg-teal-600 flex items-center transition-colors duration-300"
        >
          <input
            accept="video/*"
            class="hidden"
            id="upload-file"
            multiple=""
            type="file"
          />
          <svg
            class="text-2xl mr-4"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M448 119c0-13.3-9.4-23-22.8-23H230.9c-2.8 0-4.3-.6-6.1-2.4l-22.5-22.5-.2-.2c-4.9-4.6-8.9-6.9-17.3-6.9H88.7C74.9 64 64 74.3 64 87v73h384v-41zM64 188h-8.3c-12.8 0-25.3 5.1-23.5 24.3C34 231.5 55.7 423 55.7 423c2.7 17.8 11.7 25 25 25h352.5c12.7 0 21-7.8 23-25 0 0 22.2-184.9 23.6-205.5 1.4-20.5-8.9-29.5-23.6-29.5H64z"
            />
          </svg>
          From Device
        </li>
      </label>
      <div
        data-testid="mock-dropbox-chooser"
      >
        <li
          class="p-4 pl-8 hover:bg-teal-600 flex items-center cursor-pointer transition-colors duration-300"
        >
          <svg
            class="text-2xl mr-4"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 528 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M264.4 116.3l-132 84.3 132 84.3-132 84.3L0 284.1l132.3-84.3L0 116.3 132.3 32l132.1 84.3zM131.6 395.7l132-84.3 132 84.3-132 84.3-132-84.3zm132.8-111.6l132-84.3-132-83.6L395.7 32 528 116.3l-132.3 84.3L528 284.8l-132.3 84.3-131.3-85z"
            />
          </svg>
          From Dropbox
        </li>
      </div>
      <a
        class="p-4 pl-8 hover:bg-teal-600 flex items-center cursor-pointer transition-colors duration-300"
        href="/"
      >
        <svg
          class="text-2xl mr-4"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          version="1.1"
          viewBox="0 0 32 32"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.649 20.427h-0.126l-0.061 0.109-3.245 5.675h14.773l0.061-0.109 3.246-5.674h-0.373zM20.573 19.075l0.062 0.108h0.126l6.536-0.028-0.187-0.322-7.199-12.471-0.126 0-6.537 0.027 0.187 0.322zM15.445 12.789l-0.063-0.109-3.292-5.646-0.186 0.322-7.201 12.47 0.064 0.109 3.292 5.648 0.186-0.323 7.138-12.363z"
          />
        </svg>
        From Google Drive
      </a>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`UploadOptions Component handles Google Drive selection 1`] = `
<DocumentFragment>
  <div
    class="absolute text-2xl top-full left-0 w-full bg-teal-500 text-center text-white rounded-lg my-px shadow-lg"
  >
    <ul>
      <label
        class="cursor-pointer"
        for="upload-file"
      >
        <li
          class="p-4 pl-8 hover:bg-teal-600 flex items-center transition-colors duration-300"
        >
          <input
            accept="video/*"
            class="hidden"
            id="upload-file"
            multiple=""
            type="file"
          />
          <svg
            class="text-2xl mr-4"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M448 119c0-13.3-9.4-23-22.8-23H230.9c-2.8 0-4.3-.6-6.1-2.4l-22.5-22.5-.2-.2c-4.9-4.6-8.9-6.9-17.3-6.9H88.7C74.9 64 64 74.3 64 87v73h384v-41zM64 188h-8.3c-12.8 0-25.3 5.1-23.5 24.3C34 231.5 55.7 423 55.7 423c2.7 17.8 11.7 25 25 25h352.5c12.7 0 21-7.8 23-25 0 0 22.2-184.9 23.6-205.5 1.4-20.5-8.9-29.5-23.6-29.5H64z"
            />
          </svg>
          From Device
        </li>
      </label>
      <div
        data-testid="mock-dropbox-chooser"
      >
        <li
          class="p-4 pl-8 hover:bg-teal-600 flex items-center cursor-pointer transition-colors duration-300"
        >
          <svg
            class="text-2xl mr-4"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 528 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M264.4 116.3l-132 84.3 132 84.3-132 84.3L0 284.1l132.3-84.3L0 116.3 132.3 32l132.1 84.3zM131.6 395.7l132-84.3 132 84.3-132 84.3-132-84.3zm132.8-111.6l132-84.3-132-83.6L395.7 32 528 116.3l-132.3 84.3L528 284.8l-132.3 84.3-131.3-85z"
            />
          </svg>
          From Dropbox
        </li>
      </div>
      <a
        class="p-4 pl-8 hover:bg-teal-600 flex items-center cursor-pointer transition-colors duration-300"
        href="/"
      >
        <svg
          class="text-2xl mr-4"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          version="1.1"
          viewBox="0 0 32 32"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.649 20.427h-0.126l-0.061 0.109-3.245 5.675h14.773l0.061-0.109 3.246-5.674h-0.373zM20.573 19.075l0.062 0.108h0.126l6.536-0.028-0.187-0.322-7.199-12.471-0.126 0-6.537 0.027 0.187 0.322zM15.445 12.789l-0.063-0.109-3.292-5.646-0.186 0.322-7.201 12.47 0.064 0.109 3.292 5.648 0.186-0.323 7.138-12.363z"
          />
        </svg>
        From Google Drive
      </a>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`UploadOptions Component handles file upload 1`] = `
<DocumentFragment>
  <div
    class="absolute text-2xl top-full left-0 w-full bg-teal-500 text-center text-white rounded-lg my-px shadow-lg"
  >
    <ul>
      <label
        class="cursor-pointer"
        for="upload-file"
      >
        <li
          class="p-4 pl-8 hover:bg-teal-600 flex items-center transition-colors duration-300"
        >
          <input
            accept="video/*"
            class="hidden"
            id="upload-file"
            multiple=""
            type="file"
          />
          <svg
            class="text-2xl mr-4"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M448 119c0-13.3-9.4-23-22.8-23H230.9c-2.8 0-4.3-.6-6.1-2.4l-22.5-22.5-.2-.2c-4.9-4.6-8.9-6.9-17.3-6.9H88.7C74.9 64 64 74.3 64 87v73h384v-41zM64 188h-8.3c-12.8 0-25.3 5.1-23.5 24.3C34 231.5 55.7 423 55.7 423c2.7 17.8 11.7 25 25 25h352.5c12.7 0 21-7.8 23-25 0 0 22.2-184.9 23.6-205.5 1.4-20.5-8.9-29.5-23.6-29.5H64z"
            />
          </svg>
          From Device
        </li>
      </label>
      <div
        data-testid="mock-dropbox-chooser"
      >
        <li
          class="p-4 pl-8 hover:bg-teal-600 flex items-center cursor-pointer transition-colors duration-300"
        >
          <svg
            class="text-2xl mr-4"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 528 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M264.4 116.3l-132 84.3 132 84.3-132 84.3L0 284.1l132.3-84.3L0 116.3 132.3 32l132.1 84.3zM131.6 395.7l132-84.3 132 84.3-132 84.3-132-84.3zm132.8-111.6l132-84.3-132-83.6L395.7 32 528 116.3l-132.3 84.3L528 284.8l-132.3 84.3-131.3-85z"
            />
          </svg>
          From Dropbox
        </li>
      </div>
      <a
        class="p-4 pl-8 hover:bg-teal-600 flex items-center cursor-pointer transition-colors duration-300"
        href="/"
      >
        <svg
          class="text-2xl mr-4"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          version="1.1"
          viewBox="0 0 32 32"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.649 20.427h-0.126l-0.061 0.109-3.245 5.675h14.773l0.061-0.109 3.246-5.674h-0.373zM20.573 19.075l0.062 0.108h0.126l6.536-0.028-0.187-0.322-7.199-12.471-0.126 0-6.537 0.027 0.187 0.322zM15.445 12.789l-0.063-0.109-3.292-5.646-0.186 0.322-7.201 12.47 0.064 0.109 3.292 5.648 0.186-0.323 7.138-12.363z"
          />
        </svg>
        From Google Drive
      </a>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`UploadOptions Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="absolute text-2xl top-full left-0 w-full bg-teal-500 text-center text-white rounded-lg my-px shadow-lg"
  >
    <ul>
      <label
        class="cursor-pointer"
        for="upload-file"
      >
        <li
          class="p-4 pl-8 hover:bg-teal-600 flex items-center transition-colors duration-300"
        >
          <input
            accept="video/*"
            class="hidden"
            id="upload-file"
            multiple=""
            type="file"
          />
          <svg
            class="text-2xl mr-4"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M448 119c0-13.3-9.4-23-22.8-23H230.9c-2.8 0-4.3-.6-6.1-2.4l-22.5-22.5-.2-.2c-4.9-4.6-8.9-6.9-17.3-6.9H88.7C74.9 64 64 74.3 64 87v73h384v-41zM64 188h-8.3c-12.8 0-25.3 5.1-23.5 24.3C34 231.5 55.7 423 55.7 423c2.7 17.8 11.7 25 25 25h352.5c12.7 0 21-7.8 23-25 0 0 22.2-184.9 23.6-205.5 1.4-20.5-8.9-29.5-23.6-29.5H64z"
            />
          </svg>
          From Device
        </li>
      </label>
      <div
        data-testid="mock-dropbox-chooser"
      >
        <li
          class="p-4 pl-8 hover:bg-teal-600 flex items-center cursor-pointer transition-colors duration-300"
        >
          <svg
            class="text-2xl mr-4"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 528 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M264.4 116.3l-132 84.3 132 84.3-132 84.3L0 284.1l132.3-84.3L0 116.3 132.3 32l132.1 84.3zM131.6 395.7l132-84.3 132 84.3-132 84.3-132-84.3zm132.8-111.6l132-84.3-132-83.6L395.7 32 528 116.3l-132.3 84.3L528 284.8l-132.3 84.3-131.3-85z"
            />
          </svg>
          From Dropbox
        </li>
      </div>
      <a
        class="p-4 pl-8 hover:bg-teal-600 flex items-center cursor-pointer transition-colors duration-300"
        href="/"
      >
        <svg
          class="text-2xl mr-4"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          version="1.1"
          viewBox="0 0 32 32"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.649 20.427h-0.126l-0.061 0.109-3.245 5.675h14.773l0.061-0.109 3.246-5.674h-0.373zM20.573 19.075l0.062 0.108h0.126l6.536-0.028-0.187-0.322-7.199-12.471-0.126 0-6.537 0.027 0.187 0.322zM15.445 12.789l-0.063-0.109-3.292-5.646-0.186 0.322-7.201 12.47 0.064 0.109 3.292 5.648 0.186-0.323 7.138-12.363z"
          />
        </svg>
        From Google Drive
      </a>
    </ul>
  </div>
</DocumentFragment>
`;
